src/legacy/core_plugins/vis_type_vega/public/services.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SavedObjectsStart } from 'kibana/public';
+import { NotificationsStart } from 'src/core/public';
+import { DataPublicPluginStart } from '../../../../plugins/data/public';
+import { createGetterSetter } from '../../../../plugins/kibana_utils/public';
+
+export const [getData, setData] = createGetterSetter<DataPublicPluginStart>('Data');
+
+export const [getNotifications, setNotifications] = createGetterSetter<NotificationsStart>(
+  'Notifications'
+);
+
+export const [getSavedObjects, setSavedObjects] = createGetterSetter<SavedObjectsStart>(
+  'SavedObjects'
```

Comment:
Maybe you could use `createKibanaUtilsCore` from `kibana_utils` here.

Reply:
I think I'd make an official best practice out of either option